sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobSasImplUtil.java

```diff
@@ -319,30 +316,55 @@ private String stringToSign(String canonicalName) {
 
     private String stringToSign(final UserDelegationKey key, String canonicalName) {
         String versionSegment = this.snapshotId == null ? this.versionId : this.snapshotId;
-        return String.join("\n",
-            this.permissions == null ? "" : this.permissions,
-            this.startTime == null ? "" : Constants.ISO_8601_UTC_DATE_FORMATTER.format(this.startTime),
-            this.expiryTime == null ? "" : Constants.ISO_8601_UTC_DATE_FORMATTER.format(this.expiryTime),
-            canonicalName,
-            key.getSignedObjectId() == null ? "" : key.getSignedObjectId(),
-            key.getSignedTenantId() == null ? "" : key.getSignedTenantId(),
-            key.getSignedStart() == null ? "" : Constants.ISO_8601_UTC_DATE_FORMATTER.format(key.getSignedStart()),
-            key.getSignedExpiry() == null ? "" : Constants.ISO_8601_UTC_DATE_FORMATTER.format(key.getSignedExpiry()),
-            key.getSignedService() == null ? "" : key.getSignedService(),
-            key.getSignedVersion() == null ? "" : key.getSignedVersion(),
-            this.authorizedAadObjectId == null ? "" : this.authorizedAadObjectId,
-            "", /* suoid - empty since this applies to HNS only accounts. */
-            this.correlationId == null ? "" : this.correlationId,
-            this.sasIpRange == null ? "" : this.sasIpRange.toString(),
-            this.protocol == null ? "" : this.protocol.toString(),
-            version,
-            resource,
-            versionSegment == null ? "" : versionSegment,
-            this.cacheControl == null ? "" : this.cacheControl,
-            this.contentDisposition == null ? "" : this.contentDisposition,
-            this.contentEncoding == null ? "" : this.contentEncoding,
-            this.contentLanguage == null ? "" : this.contentLanguage,
-            this.contentType == null ? "" : this.contentType
-        );
+        if (VERSION.compareTo(BlobServiceVersion.V2019_12_12.getVersion()) <= 0) {
```

Comment:
So now are we going to support all service versions in BlobServiceVersion?

Reply:
yes. If we're giving env/props switch for this we should. Fortunately string to sign doesn't have that many variants - so far just two for all the SV we added so far.